### DISCUSSION TITLE
Merge two small generated tiles

### DISCUSSION THREAD

**Discussion by leamsigc-reflect** [openmaptiles/openmaptiles#1381] [Q&A]
Question:

I generate two small tiles:
- Luxembourg 
- Macedonia

In the data folder,that is map to the docker container of the Tileserver-GL  I have:
- Luxembourg.mbtiles 
- Macedonia.mbtiles

But when I zoom in to the map, only one is available 
- Luxembourg

So is necesarry to merge the two tiles in to a single tiles.mbtiles file ?

**Comment by dwightgunning** [comment]
This [issue](https://github.com/maptiler/tileserver-gl/issues/368) suggests tileserver-gl can only serve a single MBTiles file.

So I had 3 ideas:
1. (unsuccessful) Import multiple OSM extracts using OpenMapTiles and generate a single MBTiles file using OpenMapTiles.
2. (successful) Merge the OSM extracts using [Osmium](https://github.com/osmcode/osmium-tool). Then import that single extract and generate a single MBTiles file using OpenMapTiles for serving.
3. (not yet attempted) Repeatedly import and generate MBTiles files using OpenMapTiles. Merge the MBTiles files with [Tippecanoe](https://github.com/felt/tippecanoe#tile-join) into a single MBTiles file for serving.

Here's that second method in more detail ...

Assuming you've downloaded country extracts using `make download area=andorra` etc.

```
osmium merge data/andorra.osm.pbf data/devon.osm.pbf data/finland.osm.pbf data/ireland-and-northern-ireland.osm.pbf -o /data/out/europetest.osm.pbf
```

I wasn't able to find a recent Docker image in Docker Hub with the latest Osmium version. So I hand-rolled one myself:

```
FROM debian:bullseye-backports
RUN apt-get update && apt-get install -y osmium-tool=1.15.0-1~bpo11+1
ENTRYPOINT ["osmium"]
```

After building the docker file with a label `osmium-tool:latest`, the docker run command looks like:

```
docker run -v {/osm-data}:/data -it osmium-tool:latest merge data/andorra.osm.pbf data/devon.osm.pbf data/finland.osm.pbf data/ireland-and-northern-ireland.osm.pbf -o /data/europetest.osm.pbf
```

That merged OSM extract that can be used with OpenMapTiles ...

... with the Quickstart ...

```
make generate-bbox-file area=europetest
./quickstart.sh europetest
```

... or `import-osm` command ...

```
make import-osm area=europetest
```
